src/server/api/routers/organizations.ts

```diff
@@ -8,10 +8,85 @@ import {
   protectedProcedure,
   publicProcedure,
 } from "@/server/api/trpc";
-import { OrganizationRole } from "@/types/types";
+import { OrganizationRole, Role } from "@/types/types";
 import { passwordSchema } from "@/types/validation";
 
 export const organizationRouter = createTRPCRouter({
+  redirectToDashboard: protectedProcedure.mutation(async ({ ctx }) => {
+    const userId = ctx.session.user.id;
+
+    const user = await ctx.db.query.user.findFirst({
+      where: (user, { eq }) => eq(user.id, userId),
+    });
+
+    if (!user) {
+      throw new TRPCError({
+        code: "NOT_FOUND",
+        message: "User not found in database",
+      });
+    }
+
+    const role = user.role as Role;
+    let redirectUrl = "/";
+
+    switch (role) {
+      case Role.ADMIN:
+        redirectUrl = "/admin/home";
+        break;
+
+      case Role.AGENCY: {
+        const userSession = await ctx.db.query.session.findFirst({
```

Comment:
using the most recent session from the db might actually cause issues if the user logs in on multiple devices. for example, 
10:00 AM - User logs in on laptop → Session A created
10:05 AM - User logs in on phone → Session B created (most recent)
10:10 AM - User makes request from laptop
What happens:
-Request from laptop sends Session A token (in cookie)
-Your query returns Session B (most recently created), and the user will be redirected based on session B's org. this is quite an edge case, but we should fix it regardless

```suggestion
case Role.AGENCY: {
  // use the session from the CURRENT request
  const activeOrgId = ctx.session.activeOrganizationId;
  
  if (!activeOrgId) {
    throw new TRPCError({
      code: "NOT_FOUND",
      message: "No active organization found for this agency user",
    });
  }
  
  // ... rest of logic
}
```